autopush/router/simple.py

```diff
@@ -137,9 +145,27 @@ def route_notification(self, notification, uaid_data):
             returnValue(RouterResponse(response_body="Delivered"))
         else:
             self.metrics.increment("router.broadcast.miss")
+            if self.udp is not None:
+                yield deferToThread(self._send_udp_wake,
+                                    self.udp)
             returnValue(RouterResponse(202, "Notification Stored"))
 
-    #############################################################
+    def _send_udp_wake(self, udp_info):
```

Comment:
I was actually thinking that there'd be a IWakeup interface, and UDP wake would be one of them. They would be called after routing has completed, before the response returns, I even added a little comment to endpoint.py highlighting where the IRouter check would go to trigger it: https://github.com/mozilla-services/autopush/blob/master/autopush/endpoint.py#L404

The main reason for keeping it 100% separate from the router, is that the current router doesn't do individual messages (webpush style), while I will shortly have a new, separate router, that does. So multiple routers might want a wake-up, and hooking it onto the end of routing (at the spot in the endpoint.py) allows them to remain separate.